File: 361/solution.js
main();

function main() {
  expect([10, 9, 2, 3, 7, 101, 18], 4);
  expect([4, 4, 4, 4, 3], 1);
}

function expect(nums, expected) {
  const actual = increasingSubsequence(nums);
  if (actual != expected) {
    console.error(`Expected increasingSubsequence([${nums}]) == ${expected}, got ${actual}`);
  }
}

function increasingSubsequence(nums) {
  let 
    bestLongest = 0,
    curLongest = 0;

  if (nums.length > 0) {
    curLongest = 1;
  }

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] > nums[i-1]) {
      curLongest++;
    } else {
      bestLongest = Math.max(bestLongest, curLongest);
      curLongest = 1;
    }
  }

  return Math.max(curLongest, bestLongest);
}

File: 361/solution.go
package main

import (
	"fmt"
)

func main() {
	expect([]int{10, 9, 2, 3, 7, 101, 18}, 4)
	expect([]int{4, 4, 4, 4, 3}, 1)
}

func expect(nums []int, expected int) {
	actual := increasingSubsequence(nums)
	if actual != expected {
		fmt.Printf("Expected increasingSubsequence(%v) == %v, got %v\n", nums, expected, actual)
	}
}

func increasingSubsequence(nums []int) int {
	var i, bestLongest, longest int

	if len(nums) > 0 {
		longest = 1
	}

	for i = 1; i < len(nums); i++ {
		if nums[i] > nums[i-1] {
			longest++
		} else {
			bestLongest = max(bestLongest, longest)
			longest = 1
		}
	}

	return max(longest, bestLongest)
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
